azurelinuxagent/ga/remoteaccess.py

```diff
@@ -0,0 +1,163 @@
+            jit_users = set()
+            for usr in all_users:
+                if usr[4] == REMOTE_ACCESS_ACCOUNT_COMMENT:
+                    jit_users.add(usr[0])     
+            for acc in self.remote_access.user_list.users:
+                raw_expiration = acc.expiration
+                account_expiration = datetime.strptime(raw_expiration, REMOTE_USR_EXPIRATION_FORMAT)
+                now = datetime.utcnow()
+                if acc.name not in jit_users and now < account_expiration:
+                    self.add_user(acc.name, acc.encrypted_password, account_expiration)
+
+    def add_user(self, username, encrypted_password, account_expiration, retry=MAX_TRY_ATTEMPT, throttle=FAILED_ATTEMPT_THROTTLE):
+        created = False
+        attempt_count = 0
+        while attempt_count < retry:
+            try:
+                attempt_count += 1
+                expiration_date = account_expiration + timedelta(days=1) - datetime.utcnow()
+                logger.verbose("add_user: Adding user {0} with expiration in {1} day(s)".format(username, expiration_date.days))
+                self.os_util.useradd(username, expiration_date.days, REMOTE_ACCESS_ACCOUNT_COMMENT)
+                created = True
+                cache_file = os.path.join(conf.get_lib_dir(), "remote_access_pwd.dat")
+                prv_key = os.path.join(conf.get_lib_dir(), TRANSPORT_PRIVATE_CERT)
+                pwd = self.cryptUtil.decrypt_secret(encrypted_password, prv_key, cache_file, None)
+                self.os_util.chpasswd(username, pwd, conf.get_password_cryptid(), conf.get_password_crypt_salt_len())
+                self.os_util.conf_sudoer(username)
+                logger.info("add_user: User '{0}' added successfully with expiration in {1} day(s)".format(username, expiration_date.days))
+                return attempt_count
+            except OSError as oe:
+                self.handle_failed_create(username, oe.strerror, created)
+            except Exception as e:
+                self.handle_failed_create(username, str(e), created)
+            time.sleep(throttle)
+        logger.warn("add_user: Unable to add user {0} after {1} attemps. Will not try again for this incarnation.".format(username, MAX_TRY_ATTEMPT))
+        return attempt_count
+
+    def handle_failed_create(self, username, error_message, created):
+        logger.error("add_user: Error adding user {0}. {1}".format(username, error_message))
+        if created:
+            try:
+                self.delete_user(username)
+            except OSError as oe:
+                logger.error("clean_failed_user_create: Failed to clean up after account creation for {0}. {1}".format(username, oe.strerror()))
+            except Exception as e:
+                logger.error("clean_failed_user_create: Failed to clean up after account creation for {0}. {1}".format(username, str(e)))
+
+    def delete_user(self, username):
+        logger.info("Deleting user {0}".format(username))
```

Comment:
> logger.info(" [](start = 7, length = 14)

I would pick one log statement or the other. I vote for the second one (line 163).

Reply:
I meant to have the first verbose, but just removed it instead.

---
In reply to: [192216795](https://github.com/Azure/WALinuxAgent/pull/1171#discussion_r192216795) [](ancestors = 192216795)